src/python/grpcio/grpc/_utilities.py

```diff
@@ -189,3 +189,35 @@ def channel_ready_future(channel: grpc.Channel) -> _ChannelReadyFuture:
     ready_future = _ChannelReadyFuture(channel)
     ready_future.start()
     return ready_future
+
+
+def first_version_is_lower(version1: str, version2: str) -> bool:
+    """
+    Compares two versions in the format '1.60.1' or '1.60.1.dev0'.
+
+    This method will be used in all stubs generated by grpcio-tools to check whether
+    the stub version is compatible with the runtime grpcio.
+
+    Args:
+        version1: The first version string.
+        version2: The second version string.
+
+    Returns:
+        True if version1 is lower, False otherwise.
+    """
+    version1_list = version1.split(".")
+    version2_list = version2.split(".")
+
+    try:
+        for i in range(3):
+            if int(version1_list[i]) < int(version2_list[i]):
+                return True
+            elif int(version1_list[i]) > int(version2_list[i]):
+                return False
+    except ValueError:
+        # Return false in case we can't convert version to int.
+        return False
+
+    # If we reach here, the versions are equal up to the compared elements.
```

Comment:
The standard format is `1.2.3dev0`, not `1.2.3.dev0`. So we won't be reaching this line of code if we have `dev0`. I think this is fine because, in general, we should not be comparing versions that have different numbers of period-delimited version segments though, so maybe just remove the comment?

Reply:
Oh, you're right. My mistake.